src/components/proxy-display/proxy-display.js

```diff
@@ -0,0 +1,62 @@
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+
+import {
+  KeyValue,
+  Icon,
+} from '@folio/stripes/components';
+
+const ProxyDisplay = ({
+  proxy,
+  proxyTypesRecords,
+  inheritedProxyId,
+}) => {
+  if (proxyTypesRecords && proxy?.id && inheritedProxyId) {
```

Comment:
I'd prefer to refactor this a bit more to use fail-fast
```
const showProxyInfo = proxyTypesRecords && proxy?.id && inheritedProxyId;

if (!showProxyInfo) {
  return <Icon icon="spinner-ellipsis" />;
}

// the rest goes here
```

It reduces nesting of important logic by one level and makes the component a bit easier to read, in my opinion